common/src/main/java/com/microsoft/hydralab/performance/PerformanceTestManagementService.java

```diff
@@ -61,14 +61,18 @@ public PerformanceInspectionResult inspect(PerformanceInspection performanceInsp
         Assert.notNull(performanceInspector, "Found no matched inspector: " + performanceInspection.inspectorType);
         ITestRun testRun = getTestRun();
         File performanceFolder = new File(testRun.getResultFolder(), "performance");
-        Assert.isTrue(performanceFolder.mkdirs(), "performanceInspection.resultFolder.mkdirs() failed in " + performanceFolder.getAbsolutePath());
-        performanceInspection.resultFolder = performanceFolder;
+        Assert.isTrue(performanceFolder.exists() || performanceFolder.mkdirs(), "performanceInspection.resultFolder.mkdirs() failed in " + performanceFolder.getAbsolutePath());
+        File inspectorFolder = new File(performanceFolder, inspectorType.getName());
+        Assert.isTrue(inspectorFolder.exists() || inspectorFolder.mkdirs(), "performanceInspection.resultFolder.mkdirs() failed in " + inspectorFolder.getAbsolutePath());
+        performanceInspection.resultFolder = inspectorFolder;
 
         PerformanceInspectionResult result = performanceInspector.inspect(performanceInspection);
 
-        Map<String, PerformanceTestResult> performanceTestResultMap = testRunPerfResultMap.putIfAbsent(getTestRun(), new HashMap<>());
+        testRunPerfResultMap.putIfAbsent(getTestRun(), new HashMap<>());
+        Map<String, PerformanceTestResult> performanceTestResultMap = testRunPerfResultMap.get(getTestRun());
         Assert.notNull(performanceTestResultMap, "performanceTestResultMap should not be null ");
-        PerformanceTestResult performanceTestResult = performanceTestResultMap.putIfAbsent(performanceInspection.inspectionKey, createPerformanceTestResult(performanceInspection));
+        performanceTestResultMap.putIfAbsent(performanceInspection.inspectionKey, createPerformanceTestResult(performanceInspection));
```

Comment:
Because we should create the PerformanceTestResult only when the inspection key does not exist. We don't need to overwrite the PerformanceTestResult if the key already exists.